### PR TITLE
implement sponsor_only flag for jtbeta cores

### PR DESCRIPTION
**Pull request by mattpannella**
added name to data_slot model
check for JTBETA data_slot for sponsor_only flag
include sponsor_only in v2 api response

for issue #397

feel free to throw this PR out, i was bored this morning and had an hour to kill so i figured i'd tinker with learning some ruby

### CODE REVIEW

File: .github/workflows/inventory_updater/lib/inventory_updater.rb [hunk 1]
```diff
@@ -89,6 +89,7 @@ def process_repository(repository)
       download_url = nil
       funding = nil
       latest_release = nil
+      sponsor_only = false
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater.rb:92]:
```suggestion
```

File: .github/workflows/inventory_updater/lib/inventory_updater.rb [hunk 2]
```diff
@@ -112,6 +113,7 @@ def process_repository(repository)
         github_repository = repository.github_repository
         funding ||= @github_service.funding(github_repository)
         latest_release ||= @github_service.latest_release(github_repository, repository.prerelease) if repository.release?
+        sponsor_only = sponsor_check(core)
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater.rb:116]:
```suggestion
        requires_license = license_check(core)
```

File: .github/workflows/inventory_updater/lib/inventory_updater.rb [hunk 3]
```diff
@@ -128,7 +130,7 @@ def process_repository(repository)
         post_content = pocket_service.get_info(core_id)
         create_post(core, post_type, post_content)
 
-        serialized_cores << serialize_core(repository, core, platform, download_url, latest_release, funding)
+        serialized_cores << serialize_core(repository, core, platform, download_url, latest_release, funding, sponsor_only)
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater.rb:133]:
```suggestion
        serialized_cores << serialize_core(repository, core, platform, download_url, latest_release, funding, requires_license)
```

File: .github/workflows/inventory_updater/lib/inventory_updater.rb [hunk 4]
```diff
@@ -138,7 +140,16 @@ def process_repository(repository)
...
 
-  def serialize_core(repository, core, platform, download_url, latest_release, funding)
+  def sponsor_check(core)
+    #add custom checks as needed i guess
+    if core.author == "jotego"
+      return core.data.data_slots.any?{|data_slot| data_slot.name == "JTBETA"}
+    end
+
+    return false
+  end
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater.rb:150]:
```suggestion
  def license_check(core)
    LICENSE_SLOT_NAMES = [
      'JTBETA' # required by jotego beta cores
    ]

    return core.data.data_slots.any? { |data_slot| LICENSE_SLOT_NAMES.include?(data_slot.name) }
  end
```

File: .github/workflows/inventory_updater/lib/inventory_updater.rb [hunk 5]
```diff
@@ -138,7 +140,16 @@ def process_repository(repository)
...
+
+  def serialize_core(repository, core, platform, download_url, latest_release, funding, sponsor_only)
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater.rb:152]:
```suggestion
  def sponsor_check(core)
    #add custom checks as needed i guess
    if core.author == "jotego"
      return core.data.data_slots.any?{|data_slot| data_slot.name == "JTBETA"}
    end

    return false
  end

  def serialize_core(repository, core, platform, download_url, latest_release, funding, requires_license)
```

File: .github/workflows/inventory_updater/lib/inventory_updater.rb [hunk 6]
```diff
@@ -147,6 +158,7 @@ def serialize_core(repository, core, platform, download_url, latest_release, fun
         'name' => repository.name,
         'prerelease' => repository.prerelease
       },
+      'sponsor_only' => sponsor_only,
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater.rb:161]:
```suggestion
      'requires_license' => requires_license,
```

File: .github/workflows/inventory_updater/lib/inventory_updater/analogue/data_slot.rb [hunk 7]
```diff
@@ -2,11 +2,12 @@
 
 module Analogue
   class DataSlot
-    attr_reader :data_slot, :required, :parameters, :filename, :extensions, :configuration
+    attr_reader :data_slot, :required, :parameters, :filename, :extensions, :configuration, :name
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater/analogue/data_slot.rb:5]:
```suggestion
    attr_reader :data_slot, :name, :required, :parameters, :filename, :extensions, :configuration
```

File: api/v2/cores.json [hunk 8]
```diff
@@ -18,6 +18,7 @@ layout: none
       "release_date": {{ core.date_release | date: "%Y-%m-%d" | jsonify }},
       "platform": {{ core.platform | jsonify }},
       "assets": {{ core.assets | jsonify }},
+      "sponsor_only": {{ core.sponsor_only | jsonify }},
```
Comment (joshcampbell191) [inline api/v2/cores.json:21]:
```suggestion
      "requires_license": {{ core.requires_license | jsonify }},
```

**Review by joshcampbell191 — CHANGES_REQUESTED** [review]
Made a few recommendations.

File: .github/workflows/inventory_updater/lib/inventory_updater.rb (hunk 1 above)
Comment (mattpannella) [reply .github/workflows/inventory_updater/lib/inventory_updater.rb:92]: ah right, dont even need this

**Comment by mattpannella** [comment]
changes all look good to me. pushed updates (i'm guessing the workflow is failing because my account doesn't have permission to the api token or something?)

File: .github/workflows/inventory_updater/lib/inventory_updater.rb [hunk 9]
```diff
@@ -138,7 +139,15 @@ def process_repository(repository)
...
-  def serialize_core(repository, core, platform, download_url, latest_release, funding)
+  def license_check(core)
+    LICENSE_SLOT_NAMES = [
+      'JTBETA' # required by jotego beta cores
+    ]
+
+    return core.data.data_slots.any? { |data_slot| LICENSE_SLOT_NAMES.include?(data_slot.name) }
+  end
+
+  def serialize_core(repository, core, platform, download_url, latest_release, funding, sponsor_only)
```
Comment (joshcampbell191) [inline .github/workflows/inventory_updater/lib/inventory_updater.rb]:
```suggestion
  def serialize_core(repository, core, platform, download_url, latest_release, funding, requires_license)
```

Comment (mattpannella) [reply .github/workflows/inventory_updater/lib/inventory_updater.rb]: missed this one before. should be updated, now

**Pull request merged**